examples/benchmark.jl

```diff
@@ -0,0 +1,402 @@
+#  Copyright 2015, Vincent Leclere, Francois Pacaud and Henri Gerard
+#  This Source Code Form is subject to the terms of the Mozilla Public
+#  License, v. 2.0. If a copy of the MPL was not distributed with this
+#  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#############################################################################
+# Benchmark SDDP algorithm upon damsvalley example
+# This benchmark includes:
+# - comparison of execution time
+# - gap between the SDDP solution and the deterministic solution
+#############################################################################
+
+srand(2713)
+push!(LOAD_PATH, "../src")
+
+using StochDynamicProgramming, JuMP
+using Clp
+
+SOLVER = ClpSolver()
+#= const SOLVER = CplexSolver(CPX_PARAM_SIMDISPLAY=0, CPX_PARAM_THREADS=4) =#
+
+const N_STAGES = 10
+
+# COST:
+const COST = -66*2.7*(1 + .5*(rand(N_STAGES) - .5))
+
+# Define dynamic of the dam:
+function dynamic(t, x, u, w)
+    return [x[1] - u[1] - u[3] + w[1], x[2] - u[2] - u[4] + u[1] + u[3]]
+end
+
+# Define cost corresponding to each timestep:
+function cost_t(t, x, u, w)
+    return COST[t] * (u[1] + u[2])
+end
+
+function final_cost(x)
+	return 0.
+end
+
+
+function constraints(t, x1, u, w)
+
+    Bu = (x1[1]<=VOLUME_MAX)
+    Bl = (x1[1]>=VOLUME_MIN)
+
+    return Bu&Bl
+
+end
+
+
+"""Solve the problem with a solver, supposing the aleas are known
+in advance."""
+function solve_determinist_problem(model, scenario)
```

Comment:
change name to solve_anticipative_problem. Deterministic can be seen as open-loop control, or optimizing over the mean.